app/src/server/tasks/evaluateTestSetEntry.task.ts

```diff
@@ -9,48 +10,58 @@ import {
   getStringsToPrune,
 } from "~/modelProviders/fine-tuned/getCompletion";
 import { countLlamaChatTokens, countLlamaChatTokensInMessages } from "~/utils/countTokens";
-import { getCompletion2 } from "~/modelProviders/fine-tuned/getCompletion-2";
+import { getCompletion2, getOpenaiCompletion } from "~/modelProviders/fine-tuned/getCompletion-2";
 import { calculateEntryScore } from "../utils/calculateEntryScore";
 import { typedDatasetEntry } from "~/types/dbColumns.types";
+import { getComparisonModelName, isComparisonModel } from "~/utils/baseModels";
 
 export type EvaluateTestSetEntryJob = {
-  fineTuneId: string;
+  modelId: string;
   datasetEntryId: string;
   skipCache?: boolean;
 };
 
 export const evaluateTestSetEntry = defineTask<EvaluateTestSetEntryJob>({
   id: "evaluateTestSetEntry",
   handler: async (task) => {
-    const { fineTuneId, datasetEntryId, skipCache } = task;
+    const { modelId, datasetEntryId, skipCache } = task;
 
-    const [fineTune, rawDatasetEntry] = await prisma.$transaction([
-      prisma.fineTune.findUnique({
-        where: { id: fineTuneId },
-      }),
-      prisma.datasetEntry.findUnique({
-        where: { id: datasetEntryId },
-      }),
-    ]);
+    const rawDatasetEntry = await prisma.datasetEntry.findUnique({
+      where: { id: datasetEntryId },
+      include: {
+        dataset: {
+          include: {
+            project: true,
+          },
+        },
+      },
+    });
 
-    if (!fineTune || !rawDatasetEntry) return;
+    if (!rawDatasetEntry?.dataset) return;
 
-    const datasetEntry = typedDatasetEntry(rawDatasetEntry);
+    let fineTune;
+    if (!isComparisonModel(modelId)) {
+      fineTune = await prisma.fineTune.findUnique({
+        where: { id: modelId },
+      });
+    }
+
+    const datasetEntry = typedDatasetEntry(omit(rawDatasetEntry, "dataset"));
```

Comment:
What does this `omit` do?

Reply:
Nothing, I removed it. I added it because I was getting a type error and wasn't sure if it was due to the rawDatasetEntry having more fields than typedDatasetEntry expected.